File: src/main/java/com/netsensia/rivalchess/engine/core/eval/PieceValue.java
package com.netsensia.rivalchess.engine.core.eval;

import com.netsensia.rivalchess.model.Piece;

public class PieceValue {

    public static int getValue(Piece piece) {
        switch (piece) {
            case PAWN: return 100;
            case KNIGHT: return 390;
            case BISHOP: return 390;
            case ROOK: return 595;
            case KING: return 30000;
            case QUEEN: return 1175;
            case NONE: return 0;
            default:
                throw new RuntimeException("Unknown piece");
        }
    }
}

File: src/main/java/com/netsensia/rivalchess/engine/core/eval/PieceValue.kt
package com.netsensia.rivalchess.engine.core.eval

import com.netsensia.rivalchess.engine.core.*
import com.netsensia.rivalchess.model.Piece

const val VALUE_PAWN = 100
const val VALUE_KNIGHT = 390
const val VALUE_BISHOP = 390
const val VALUE_ROOK = 595
const val VALUE_KING = 30000
const val VALUE_QUEEN = 1175

fun pieceValue(piece: Piece): Int {
    return when (piece) {
        Piece.PAWN -> VALUE_PAWN
        Piece.KNIGHT -> VALUE_KNIGHT
        Piece.BISHOP -> VALUE_BISHOP
        Piece.ROOK -> VALUE_ROOK
        Piece.KING -> VALUE_KING
        Piece.QUEEN -> VALUE_QUEEN
        Piece.NONE -> 0
    }
}

fun pieceValue(bitboardType: Int): Int {
    return when (bitboardType) {
        BITBOARD_WP, BITBOARD_BP -> 100
        BITBOARD_WN, BITBOARD_BN -> 390
        BITBOARD_WB, BITBOARD_BB -> 390
        BITBOARD_WR, BITBOARD_BR -> 595
        BITBOARD_WK, BITBOARD_BK -> 30000
        BITBOARD_WQ, BITBOARD_BQ -> 1175
        else -> 0
    }
}
